### DISCUSSION TITLE
Removal of {{interface}}, {{ifattribute}} and {{ifmethod}} macros

### DISCUSSION THREAD

**Discussion by teoli2003** [mdn/mdn-community#67] [Localization] [answered]
Hey @mdn/localization-team-leads, 

The `{{interface}}`, `{{ifattribute}}` and `{{ifmethod}}` macros are removed from mdn/content: mdn/content#, mdn/content#15848 and mdn/content#15850
and marked as deprecated in mdn/yari: mdn/yari#6011, mdn/yari#https://github.com/mdn/yari/pull/6205

When you have a moment, we would appreciate it if you could start the process of removing the macro from your area of `mdn/translated-content`.

**Thank you** for your continued work to ensure the health of our translated content.

## `{{interface}}`

`{{interface("name")}}` should become `name`.

I found 105 occurrences left in `mdn/translated-content` (`rg -i "\{ *(interface)" -l |wc -l`).

## `{{ifmethod}}`

`{{ifmethod("if", "name")}}` should become `if.name()`.

I found 33 occurrences left in `mdn/translated-content` (`rg -i "\{ *if(method)" -l |wc -l`).

## `{{ifattribute}}`

`{{ifmethod("if", "name")}}` should become `if.name`.

I found 10 occurrences left on `mdn/translated-content` (`rg -i "\{ *if(attribute)" -l |wc -l`).

-- 
Teoli, killing Kumascript, one macro at a time (ok, three here)

**Comment by yin1999** [comment]
`{{interface}}` already removed from `l10n-zh`, created mdn/translated-content#5533 for removing `{{ifmethod}}` and `{{ifattribute}}` from `l10n-zh`.

**Comment by SphinxKnight** [comment]
Regarding `fr`, @cw118 already took care of `{{interface}}` in https://github.com/mdn/translated-content/pull/5194 (also see https://github.com/mdn/translated-content/issues/5193 which she wrote for coordination in the first place :pray: :bow: :100: )

Let's get rid of the rest :fire:

**Reply by SphinxKnight** [reply]
- `{{ifiattribute}}` https://github.com/mdn/translated-content/pull/5537
- `{{ifmethod}}` https://github.com/mdn/translated-content/pull/5538

**Comment by lex111** [comment]
Done it for Ru locale https://github.com/mdn/translated-content/pull/5571

**Comment by mfuji09** [comment]
Created PR https://github.com/mdn/translated-content/pull/5694 for `ja`.

**Comment by teoli2003** [comment]
Thanks all for the good work. We have made great progress already:
- `{{interface}}` is gone.
- `{{ifmethod}}` has only few occurrences left:
  - files/de/web/api/window/alert/index.html

  - files/es/web/api/window/alert/index.html
  - files/es/web/api/web_workers_api/using_web_workers/index.html
  - files/es/web/api/htmlcanvaselement/index.html
  - files/es/mdn/structures/macros/other/index.html
  - files/es/mozilla/firefox/releases/9/updating_add-ons/index.html
 
  - files/pt-br/web/api/window/innerheight/index.html
  - files/pt-br/web/api/window/alert/index.html

- `{{ifattribute}}` has only 1 occurrence left:
  - files/es/mozilla/firefox/releases/9/updating_add-ons/index.html

**Reply by SphinxKnight** [reply]
I'll take `de` files

**Reply by SphinxKnight** [reply]
https://github.com/mdn/translated-content/pull/6313 done for `ifmethod` for `de`

**Comment by JuanVqz** [comment]
Have opened an issue to solve this for `es`
https://github.com/mdn/translated-content/issues/6337

**Reply by JuanVqz** [reply]
just for the record, we are almost done only one more file to do. :+1:

**Reply by teoli2003** [reply]
Yes! Thanks all for the good work. We have made great progress already:
- `{{interface}}` is gone.
- `{{ifmethod}}` has only three occurrences left:

  - files/es/mozilla/firefox/releases/9/updating_add-ons/index.html
 
  - files/pt-br/web/api/window/innerheight/index.html
  - files/pt-br/web/api/window/alert/index.html

- `{{ifattribute}}` has only 1 occurrence left (same file as for `{{ifmethod}}`!):
  - files/es/mozilla/firefox/releases/9/updating_add-ons/index.html 

With the last file for Spanish having an issue opened.

Thanks a lot!

**Reply by SphinxKnight** [reply]
@mdn/yari-content-pt-br: the last two files with the `{{ifmethod}}` are:

- files/pt-br/web/api/window/innerheight/index.html
- files/pt-br/web/api/window/alert/index.html

**Comment by SphinxKnight** [comment, marked as answer]
As of today, only 2 occurrences of `{{ifmethod}}` remain: I opened https://github.com/mdn/translated-content/pull/7984

Both `{{ifmethod}}` and `{{interface}}` are gone from mdn/translated-content

**Reply by teoli2003** [reply]
Well done! We can remove all three macros!
